dali/base/dadfs.cpp

```diff
@@ -2738,6 +2738,20 @@ class CDistributedFileBase : implements INTERFACE, public CInterface
         return *t;
     }
 
+    IPropertyTree &queryHistory()
+    {
+        IPropertyTree *t = root->queryPropTree("History");
+        if (!t)
+            t = root->setPropTree("History",createPTree("History")); // takes ownership
```

Comment:
I'm not sure. I suppose because it's there and in use it hasn't caused problems, but not sure it's correct either.

As long as it doesn't expect the change committed it may be okay in both cases and/or if it's inside a lock.

Are there any cases you use queryHistory() you expect it to be committed?

I _think_ queryAttributes() and queryHistory() in the interface should only every be used for reading purposes. They return a IPT which could be changed, but it will have no effect.

It would be good to confirm and add comments.